test/e2e/tests/app_test.go

```diff
@@ -44,13 +44,17 @@ func TestApp_Hash(t *testing.T) {
 		require.NoError(t, err)
 		require.NotEmpty(t, info.Response.LastBlockAppHash, "expected app to return app hash")
 
-		block, err := client.Block(ctx, &info.Response.LastBlockHeight)
+		status, err := client.Status(ctx)
 		require.NoError(t, err)
-		require.EqualValues(t, info.Response.LastBlockAppHash, block.Block.AppHash.Bytes(),
-			"app hash does not match last block's app hash")
 
-		status, err := client.Status(ctx)
+		block, err := client.Block(ctx, &info.Response.LastBlockHeight)
 		require.NoError(t, err)
+
+		if info.Response.LastBlockHeight == block.Block.Height {
+			require.EqualValues(t, info.Response.LastBlockAppHash, block.Block.AppHash.Bytes(),
+				"app hash does not match last block's app hash")
+		}
```

Comment:
As has been discussed, this more or less is just testing the `/Block` endpoint which should ideally be covered in unit tests. In any case, if there is something wrong with the method, this will fail silently (or pass even) which I'm not sure is the desired behavior.

Reply:
I think if we're worried about `/block` sometimes returning a different block than the one you ask for, we should write a unittest for it and not do it here (and it looks like we basically do this here https://github.com/tendermint/tendermint/blob/master/internal/store/store_test.go#L467-L493, and elsewhere in the e2e tests https://github.com/tendermint/tendermint/blob/master/test/e2e/tests/block_test.go#L88. 

I suspect that we could turn this back on, but also I'm fine with not because I thin if this fails, this is a very confusing place for the error (we could fix it by making an assertion about heights being equal earlier, though because any failure here would ends up being intermittent and difficult to reproduce, I'd rather avoid it.)